src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftPermissionFilter.java

```diff
@@ -78,25 +126,61 @@ public void doFilter(ServletRequest request, ServletResponse response,
 							s.setAttribute(OAuthSession.SESSION_NAME + LAST_SELF_SAR_POLL_TIME, new Long(System.currentTimeMillis()));
 						}
 						
-						long interval = SELF_SAR_POLL_INTERVAL;
-						String var = EnvVars.masterEnvVars.get(OPENSHIFT_PERMISSIONS_POLL_INTERVAL);
-						if (var != null) {
-							try {
-								interval = Long.parseLong(var);
-							} catch (Throwable t) {
-								
-							}
-						}
 						if (updated || (System.currentTimeMillis() - lastPermissionPoll.longValue() > (interval * 1000))) {
 							OpenShiftOAuth2SecurityRealm secRealm = (OpenShiftOAuth2SecurityRealm) Jenkins.getInstance().getSecurityRealm();
-							secRealm.updateAuthorizationStrategy(oauth);
+							secRealm.updateAuthorizationStrategy(oauth.getCredential());
 							s.setAttribute(OAuthSession.SESSION_NAME + LAST_SELF_SAR_POLL_TIME, new Long(System.currentTimeMillis()));
 						}
 					} catch (Throwable t) {
 						OpenShiftOAuth2SecurityRealm.LOGGER.log(Level.SEVERE, "filter", t);
 					}
 				}
+			} else {
+			    // support for non-browser, like curl, access to jenkins with openshift oauth security;
+			    // by choice, not storing auth in http session (remember, no browser) or anything like that;
+			    // want the token provided on each access 
+			    try {
+			        String enabled = EnvVars.masterEnvVars.get(OPENSHIFT_ACCESS_VIA_BEARER_TOKEN);
+			        if (enabled == null || !enabled.equalsIgnoreCase("false")) {
+	                    String authHdr = httpRequest.getHeader("Authorization");
+	                    if (authHdr != null && authHdr.length() > 0 && authHdr.startsWith("Bearer")) {
+	                        String[] words = authHdr.split(" ");
+	                        if (words.length > 1) {
+	                            String token = words[1];
+	                            
+	                            BearerCacheEntry entry = bearerCache.get(token);
+	                            if (entry == null) {
+                                    entry = new BearerCacheEntry();
+                                    bearerCache.put(token, entry);
```

Comment:
Yes, "it's ok that the lookup returns an entry w/o a valid token".  cacheEntry.token can be null if the user supplied token previously failed the openshift oauth flows.  If so, and the time of the last check is still withing the polling interval, we'll immediate send the 401 without making another oauth req against openshift.  Then, when we've waited past the polling interval, we'll retry the openshift oauth reqs with the token supplied on the http req, and see if it is valid.